common/src/com/thoughtworks/go/buildsession/TestCommandExecutor.java

```diff
@@ -37,6 +37,14 @@ public boolean execute(BuildCommand command, BuildSession buildSession) {
             return !left.equals(captureSubCommandOutput(command, buildSession));
         }
 
+        if ("-in".equals(flag)) {
+           return captureSubCommandOutput(command, buildSession).contains(left);
+        }
+
+        if ("-nin".equals(flag)) {
+            return !captureSubCommandOutput(command, buildSession).contains(left);
+        }
+
```

Comment:
@arvindsv I opened a pull request for these changes on the golang agent (https://github.com/gocd-contrib/gocd-golang-agent/pull/47)